src/uwtools/api/config.py

```diff
@@ -103,26 +97,7 @@ def realize(
     dry_run: bool = False,
 ) -> bool:
     """
-    Realize an output config based on an input config and an optional values-providing config.
-
-    If no input is specified, ``stdin`` is read. If no output is specified, ``stdout`` is written
-    to. A ``dict`` may also be provided as an input value. When a filename is specified for an input
-    or output, its format will be deduced from its extension, if possible. This can be overriden by
-    specifying the format explicitly, and it is required to do so for reads from ``stdin`` or writes
-    to ``stdout``, as no attempt is made to deduce the format of streamed data.
-
-    If ``values_needed`` is ``True``, a report of values needed to realize the config is logged. In
-    ``dry_run`` mode, output is written to ``stderr``.
-
-    :param input_config: Input config file (``None`` or unspecified => read ``stdin``)
-    :param input_format: Format of the input config
-    :param output_file: Output config file (``None`` or unspecified => write to ``stdout``)
-    :param output_format: Format of the output config
-    :param values: Source of values used to modify input
-    :param values_format: Format of values when sourced from file
-    :param values_needed: Report complete, missing, and template values
-    :param dry_run: Log output instead of writing to output
-    :return: ``True``
+    NB: This docstring is dynamically replaced: See realize.__doc__ definition below.
```

Comment:
Switch to dynamic docstrings so that the list of recognized extensions can be interpolated into the docstring. This technique was already in use in module `uwtools.api.forecast` to that the supported models (admittedly, only one currently) could similarly be interpolated into the docstring for the `run()` function.